### DISCUSSION TITLE
Using RegEx within the 1.9.1 GUI

### DISCUSSION THREAD

**Discussion by dorvashmr** [vdemichev/DiaNN#1209] [General]
Hi Vadim,

Thank you for the incredible improvements in the new version.
I had a really naïve question about the option to use RegEx in the GUI. Could you please provide an example of how it can be used, for instance, to replace the output path or the input files? The major issue is not the "regex" expressions, but how to use them in diann.

Thank you very much.

Cheers,
Reza

**Comment by vdemichev** [comment]
Hi Reza,

The purpose of this is to allow to take a large pipeline and easily transform it to work on another PC, where the same raw data files might be in different locations and output needs to be directed to different folders. Also allows to target pipeline to a specific DIA-NN version.

The syntax is --regex (typed in the RegEx field) followed by two strings in quotes, the first is the search string (can be a regular expression), the second is a relace string (can also be a regular expression). You can have multiple --regex commands in the field - all will be applied to all pipeline steps and all file location strings in those upon pressing the RegEx button. Add --ignore-case before adding any --regex commands to ignore letter case.

Examples:
```
--regex "D:" "C:" 
--regex "diann.exe" "C:/DIA-NN/1.9.1/diann.exe"
--ignore-case --regex "c:/temp" "c:/temp/1.9.1/" --regex "diann.exe" "C:/dia-nn/1.9.1/diann.exe"
```

Best,
Vadim

**Reply by dorvashmr** [reply]
Hi Vadim,

It's incredibly handy. Also perfect descriptions. 
Thank you.

Cheers,
Reza
